themes/default/content/docs/languages-sdks/python/python-blocking-async.md

```diff
@@ -0,0 +1,199 @@
+  if res.returncode is not 0:
+    raise Error(f'my-cli returned {res.returncode}: {res.stderr}')
+  return res.stdout
+
+# Use asyncio to call the blocking_operation function on another thread
+# This returns a coroutine
+mycli_coro = asyncio.to_thread(blocking_operation, 'bar')
+# The coroutine is awaitable and can be converted into a pulumi.Output
+mycli_output = pulumi.Output.from_input(mycli_coro)
+```
+
+If an output needs to be transformed, for example if a blocking call returns a
+complex object from which a value needs to be extracted (itself as a
+`pulumi.Output`) then
+[`pulumi.Output.apply`](/docs/reference/pkg/python/pulumi#pulumi.Output.apply) can
+be used to transform an output value
+
+### Asynchronous Code
+
+Explicitly async code can be used in exactly the same way---`pulumi.Output.from_input` can convert any awaitable value into a `pulumi.Output`.
+To use the same example as above with an equivalent asynchronous implementation
+
+```python
+import asyncio
+import shlex
+
+async def async_operation(foo: str) -> str:
+  """
+  An example async function that calls a subprocess and captures its output
+  """
+  proc = await asyncio.create_subprocess_shell(
+    f'my-cli {shlex.quote(foo)}',
+    stdout=asyncio.subprocess.PIPE
+    stderr=asyncio.subprocess.PIPE)
+
+  stdout, stderr = await proc.communicate()
+		
+  if proc.returncode is not 0:
+    raise Error(f'my-cli returned {proc.returncode}: {stderr.decode()}')
+  return stdout.decode
+
+# Calling an async function directly (without the await keyword) returns a
+# coroutine
+mycli_coro = async_operation('bar')
+# The coroutine is awaitable and can be converted into a pulumi.Output
+mycli_output = pulumi.Output.from_input(mycli_coro)
+```
+
+## Alternatives
+
+The examples above both use calling a local process to demonstrate a pattern that is useful in Python Pulumi programs. This applies to other blocking and async requirements too, for example if you need to call a REST API you might use `requests.get` or an async equivalent such as `httpx.AsyncClient.get`
+
+For running other processes, you can stay entirely within the Pulumi programming model by using the [Command](/registry/packages/command) package. With this package both of the examples above can be
+simplified to the following
+
+```python
+import shlex
+from pulumi_command import local
+
+def call_my_cli(foo: str) -> Output[str]:
+    """
+    As example function that uses the Command package to run a local process
+    """
+    my_cli = local.Command('my-cli', create=f'my-cli {shlex.quote(foo)})')
+    return my_cli.stdout
+
+# The return from call_my_cli is already a pulumi.Output
+mycli_output = call_my_cli('bar')
+```
+
+If the input to the function needs to be a `pulumi.Output` itself---_i.e._ if it is the result of creating another resource, or calling another command, then we can use [`pulumi.Output.concat`](/docs/reference/pkg/python/pulumi#pulumi.Output.concat) as follows
```

Comment:
nit: is it normal for us to go into a code block without a colon?

Reply:
I'll add colons